pkg/apis/planetscale/v2/vitessshard_types.go

```diff
@@ -305,37 +305,68 @@ type VitessShardStatus struct {
 	// Cells is a list of cells in which any tablets for this shard are deployed.
 	Cells []string `json:"cells,omitempty"`
 
-	// HasMaster is a condition indicating whether the Vitess topology
```

Comment:
We can't get rid of these, unfortunately, because it's a breaking change. Upon upgrading the operator, existing clusters may be disrupted because one controller looks for the new conditions but they haven't been set yet by another controller. That could lead to bad things like thinking there is no master yet or that we haven't taken the initial backup yet.

The reason these were not full condition structs to begin with is because some in the k8s project considered status condition structs to be a mistake and that plain fields were better for anything that isn't "dynamic" - fields that you expect to always be there. I was trying to learn from those mistakes.

The reason we're now adding a conditions list is because we want to support dynamic conditions - things added by outside controllers that we allow and preserve. It's sort of like a more structured and narrow form of annotations.

Reply:
Brought them all back and reverted code back to depending directly on these fields https://github.com/planetscale/vitess-operator/pull/31/commits/55d6ac920b9cd1e39e4d37bcda6d56a073a4c317.